packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java

```diff
@@ -360,12 +365,14 @@ public void setRemoveClippedSubviews(boolean removeClippedSubviews) {
         mAllChildren[i] = child;
         child.addOnLayoutChangeListener(mChildrenLayoutChangeListener);
       }
+      mChildrenRemovedWhileTransitioning = new HashSet<>();
```

Comment:
Let's allocate this lazily? Most apps won't have transitioning views and shouldn't pay for this.

Reply:
https://github.com/facebook/react-native/pull/47634/commits/c3da8d0987ce97bc5bb21915c25b43c3d4f76d3f -> now we allocate the hash set only in case a view has been removed while transitioning.